src/main/java/com/uid2/client/InputUtil.java

```diff
@@ -0,0 +1,127 @@
+package com.uid2.client;
+
+import java.util.Base64;
+
+public class InputUtil { //from https://github.com/IABTechLab/uid2-operator/blob/master/src/main/java/com/uid2/operator/service/InputUtil.java
+  /**
+   * @param phoneNumber a phone number in any format
+   * @return whether phoneNumber is normalized, which is a requirement for {@link TokenGenerateInput#fromPhone}
+   */
+  public static boolean isPhoneNumberNormalized(String phoneNumber) {
+    //from https://github.com/IABTechLab/uid2-operator/blob/14de3733e72f72adf1d9af7091dee03ea9cdb5b2/src/main/java/com/uid2/operator/service/InputUtil.java#L80
+    final int MIN_PHONENUMBER_DIGITS = 10;
+    final int MAX_PHONENUMBER_DIGITS = 15;
+
+    // normalized phoneNumber must follow ITU E.164 standard, see https://www.wikipedia.com/en/E.164
+    if (phoneNumber == null || phoneNumber.length() < MIN_PHONENUMBER_DIGITS)
+      return false;
+
+    // first character must be '+' sign
+    if ('+' != phoneNumber.charAt(0))
+      return false;
+
+    // count the digits, return false if non-digit character is found
+    int totalDigits = 0;
+    for (int i = 1; i < phoneNumber.length(); ++i)
+    {
+      if (!InputUtil.isAsciiDigit(phoneNumber.charAt(i)))
+        return false;
+      ++totalDigits;
+    }
+
+    return totalDigits >= MIN_PHONENUMBER_DIGITS && totalDigits <= MAX_PHONENUMBER_DIGITS;
+  }
+
+  static String normalizeEmailString(String email) {
```

Comment:
Is this intended to be marked public?

Reply:
no, it's called implicitly when using an email address input. And an identical function is called in the Operator if an unhashed email is sent in the API